Replicant support

DarkmatterVale opened the pull request:
This PR pretty much redoes most of Tipster, giving it replicant support.

DarkmatterVale committed:
Allow use to get previous & next tip, and set delay

DarkmatterVale committed:
Add replicant functionality

DarkmatterVale committed:
Attempt to convert Tipster into a View

DarkmatterVale committed:
Moved all components into Tipster class

DarkmatterVale committed:
Clean up Window class

DarkmatterVale committed:
Got layout mostly working

DarkmatterVale committed:
Finish updating the layout & re-enabling all previous functionality

DarkmatterVale committed:
Add Replicant support

DarkmatterVale committed:
Try 2 at replicant support...

DarkmatterVale committed:
Replicant try 3....

DarkmatterVale committed:
Replicant can now be created

DarkmatterVale committed:
Move replicant actions

DarkmatterVale committed:
Clean up code

DarkmatterVale committed:
More code editing to attempt to get replicant working

DarkmatterVale committed:
Finally got replicant support working

DarkmatterVale committed:
Finalize code, except for one bug

DarkmatterVale committed:
Layout looks a little better now, but still not ideal

DarkmatterVale committed:
Fully complete the replicant

DarkmatterVale committed:
More code cleanup, add AboutWindow to replicant

DarkmatterVale committed:
Added Icon, cleaned up code

DarkmatterVale committed:
Added a destructor

DarkmatterVale committed:
Clean up delay message code

pulkomandy commented on Tipster/MainWindow.cpp:
```diff
@@ -102,35 +102,39 @@ MainWindow::MessageReceived(BMessage* msg)
+			fTipsterView->DisplayPreviousTip();
+			break;
+		}
+		case DELAY_30_S:
```
Doesn't this code feel a bit repetitive?
I suggest you have a look at BMessage->AddInt32 and BMessage->FindInt32, which will allow you to use a single message constant with different values.

pulkomandy commented on Tipster/Tipster.cpp:
```diff
@@ -33,14 +41,144 @@ enum
+	BRect rect(Bounds());
+	rect.top = rect.bottom - 7;
+	rect.left = rect.right - 7;
+	BDragger* dragger = new BDragger(rect, this,
```
You should use the constructor without a "rect" argument, as you are in a layouted view.

DarkmatterVale replied:
It does not display properly if I remove the rect. In fact, it does not display at all.

DarkmatterVale committed:
Fix capitalization errors

DarkmatterVale committed:
Instantiate fPreviousTip preventing a possible bug

DarkmatterVale committed:
Fix bug

DarkmatterVale committed:
Remove unused variable

humdingerb commented:
Guys, shouldn't ValeT's PR be merged before Hannah works on localization?

mmuman commented:
Indeed.

mmuman commented:
I'll just merge as is, he can always open a new one for more fixes.
I checked, #5 can be rebased with a single conflict, good occasion to learn git rebase.

mmuman commented:
@DarkmatterVale please make sure you addressed all comments, if not open a new PR.

DarkmatterVale commented:
@mmuman I believe I have addressed all comments.

The only comment that I am "ify" on is the one regarding the way I instantiate the BDragger. As I stated in  my answer to that comment, I am unable to view the dragger at all if I do not instantiate it the way I do.
